charts/keycloak/templates/_helpers.tpl

```diff
@@ -65,7 +65,7 @@ Create a default fully qualified app name for the postgres requirement.
 */}}
 {{- define "keycloak.postgresql.fullname" -}}
 {{- $postgresContext := dict "Values" .Values.postgresql "Release" .Release "Chart" (dict "Name" "postgresql") -}}
-{{ include "postgresql.fullname" $postgresContext }}
+{{ include "postgresql.primary.fullname" $postgresContext }}
```

Comment:
on local install I get the error message. Please fix this or show me what i did wrong.
```
❯ helm install my-test4 .
Error: template: keycloak/templates/statefulset.yaml:58:48: executing "keycloak/templates/statefulset.yaml" at <include "keycloak.postgresql.fullname" .>: error calling include: template: keycloak/templates/_helpers.tpl:68:3: executing "keycloak.postgresql.fullname" at <include "postgresql.primary.fullname" $postgresContext>: error calling include: template: no template "postgresql.primary.fullname" associated with template "gotpl"
```

Reply:
That's a hard one, since I cannot reproduce your error:
```
$ git status
On branch bump-postgres-in-keycloak
$ helm install my-test42 .
NAME: my-test42
LAST DEPLOYED: Thu Apr 15 20:19:59 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
***********************************************************************
*                                                                     *
*                Keycloak Helm Chart by codecentric AG                *
*                                                                     *
***********************************************************************

Keycloak was installed with a Service of type ClusterIP

Create a port-forwarding with the following commands:

export POD_NAME=$(kubectl get pods --namespace default -l "app.kubernetes.io/name=keycloak,app.kubernetes.io/instance=my-test42" -o name)
echo "Visit http://127.0.0.1:8080 to use your application"
kubectl --namespace default port-forward "$POD_NAME" 8080
```
```

Reply:
My bad, i forgot to run `helm dep build`. 

But the old and the new version are not compatible out of the box. When upgrading an old release I get the error `Error: UPGRADE FAILED: cannot patch "test-postgresql" with kind StatefulSet: StatefulSet.apps "test-postgresql" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden`

This means this change requires a major version upgrade of the chart as well as upgrade instructions. In this case typically a `kubectl delete --cascade=false sts test-postgresql`, `helm upgrade`, ` kubectlk delete olad database pods`

Reply:
Thanks for your feedback and testing.

This is resolved per my latest force push. Your procedure (delete without cascade) is (IMHO) more complex than necessary: just deleting the STS will keep the PVC in place. This is what most helm charts I use recommend as upgrade path in this situation, also.